### PR TITLE
Feature/rename chef solo

### PR DESCRIPTION
**Pull request by dereklwood**
Renaming the Chef Automator plugin to the Chef Solo Automator plugin, and corresponding automatortype from "chef" to "chef-solo".  

This updates:
- [x] provisioner plugin code
- [x] config defaults and examples
- [x] documentation
- [x] unit tests for UI and Server

### CODE REVIEW

File: config/defaults/services/nodejs.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "provisioner": {
     "actions": {
       "install": {
-        "type":"chef",
+        "type":"chef-solo-solo",
```
Comment (wolf31o2) [inline config/defaults/services/nodejs.json]: There can be only one!

File: docs/docs/source/guide/admin/chef-solo-automator-plugin.rst [hunk 2]
```diff
@@ -154,39 +154,39 @@ Each Loom Automator plugin is responsible for implementing a bootstrap method in
...
 
 
 Adding your own Cookbooks
 =========================
 **Cookbook requirements**
 
-Since the Chef Automator plugin is implemented using chef-solo, the following restrictions apply:
+Since the Chef Solo Automator plugin is implemented using chef-solo, the following restrictions apply:
 
 	* No chef search capability
```
Comment (wolf31o2) [inline docs/docs/source/guide/admin/chef-solo-automator-plugin.rst]: Not a part of the diff, exactly, but let's capitalize Chef

File: docs/docs/source/guide/admin/chef-solo-automator-plugin.rst [hunk 3]
```diff
@@ -207,7 +207,7 @@ all the nodes in the cluster. It then simply utilizes the community "hostsfile"
 each node.
 
 The example loom service definition invoking this cookbook is called "hosts". It simply sets up a "configure" service
```
Comment (wolf31o2) [inline docs/docs/source/guide/admin/chef-solo-automator-plugin.rst]: This is done by base, now, not hosts.

**Comment by wolf31o2** [comment]
A couple comments... only 1 error that needs to be fixed. Otherwise, LGTM.

**Comment by dereklwood** [comment]
addressed comments... needed to add a quick section on recipe loom_base, which should be reviewed: https://github.com/continuuity/loom/commit/66ea4dac312f21de3d2422165a143ffbe8ad6696

**Comment by wolf31o2** [comment]
New loom_base looks good. :+1:

**Pull request merged**